### ISSUE TITLE
why the lightGBM use rmse for classification module?

### ISSUE THREAD

**Issue by Green-16** [microsoft/LightGBM#5397]
In my opinion , classification module should use metrics  such as AUC , KS  ,confusion matrix and so on . 

https://github.com/microsoft/LightGBM/blob/7de8d8cf3eb50fcfd910cdc7cc4f1ede94fb4358/examples/python-guide/advanced_example.py#L87

**Comment by jameslamb** [comment]
Thanks very much for pointing this out!

It looks like that example has been using RMSE for a binary classification model for 5+ years, since https://github.com/microsoft/LightGBM/pull/371.

I agree with you that it's a little surprising to see RMSE in a binary classification example. Would you like to submit a pull request updating that example to use a different metric?

If you do, please:

* update all places in `advanced_example.py`
* use a metric from `sklearn.metrics` (not one you write yourself)
* only use a single metric (this portion of this example is intended to show how to save and load models, not teach people all possible ways to evaluate a binary classification model)

**Comment by Green-16** [comment]
ok, I can do that.

**Comment by Green-16** [comment]
I have  submit the pull request . @jameslamb   https://github.com/microsoft/LightGBM/pull/5440
